server/migrations/20231129175445-updateResultsUnexpectedBehaviorSeverityAndText.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+    async up(queryInterface) {
+        return queryInterface.sequelize.transaction(async transaction => {
+            const [testPlanRuns] = await queryInterface.sequelize.query(
+                `select id, "testResults" from "TestPlanRun"`,
+                {
+                    transaction
+                }
+            );
+
+            for (let testPlanRun of testPlanRuns) {
+                let needsUpdate = false;
+                let testResultsToUpdate = [];
+
+                for (let testResult of testPlanRun.testResults) {
+                    for (let scenarioResult of testResult.scenarioResults) {
+                        if (scenarioResult.unexpectedBehaviors) {
+                            for (let unexpectedBehavior of scenarioResult.unexpectedBehaviors) {
+                                // Will only apply if unexpectedBehavior.id = OTHER
+                                if (
+                                    unexpectedBehavior.otherUnexpectedBehaviorText
+                                )
+                                    unexpectedBehavior.unexpectedBehaviorText =
+                                        unexpectedBehavior.otherUnexpectedBehaviorText;
+
+                                // Default severity to Moderate
+                                if (!unexpectedBehavior.severity)
+                                    unexpectedBehavior.severity = 'MODERATE';
+
+                                // Default the text to N/A. Text content is required
+                                if (!unexpectedBehavior.unexpectedBehaviorText)
+                                    unexpectedBehavior.unexpectedBehaviorText =
+                                        'N/A';
+                            }
+                            needsUpdate = true;
+                        }
+                    }
+
+                    testResultsToUpdate.push(testResult);
```

Comment:
Does the test result need to pushed outside of the if statement, given that `needsUpdate` is being set?

Reply:
What kind of change were you thinking?

Reply:
With something like the following. Does it read a bit better?

```javascript
for (let testPlanRun of testPlanRuns) {
    let needsUpdate = testPlanRun.testResults.some(testResult =>
        testResult.scenarioResults.some(
            scenarioResult => !!scenarioResult.unexpectedBehaviors
        )
    );
    let testResultsToUpdate = [];

    if (needsUpdate) {
        for (let testResult of testPlanRun.testResults) {
            for (let scenarioResult of testResult.scenarioResults) {
                if (scenarioResult.unexpectedBehaviors) {
                    for (let unexpectedBehavior of scenarioResult.unexpectedBehaviors) {
                        // Will only apply if unexpectedBehavior.id = OTHER
                        if (
                            unexpectedBehavior.otherUnexpectedBehaviorText
                        )
                            unexpectedBehavior.unexpectedBehaviorText =
                                unexpectedBehavior.otherUnexpectedBehaviorText;

                        // Default severity to Moderate
                        if (!unexpectedBehavior.severity)
                            unexpectedBehavior.severity =
                                'MODERATE';

                        // Default the text to N/A. Text content is required
                        if (
                            !unexpectedBehavior.unexpectedBehaviorText
                        )
                            unexpectedBehavior.unexpectedBehaviorText =
                                'N/A';
                    }
                }
            }

            testResultsToUpdate.push(testResult);
        }

        await queryInterface.sequelize.query(
            `update "TestPlanRun" set "testResults" = ? where id = ?`,
            {
                replacements: [
                    JSON.stringify(testResultsToUpdate),
                    testPlanRun.id
                ],
                transaction
            }
        );
    }
}
```